### ISSUE TITLE
PriceTaker Model Checklist

### ISSUE THREAD

**Issue by adam-a-a** [IDAES/idaes-pse#1449]
Running TODO for Pricetaker class:

- [ ] Add demo/tutorial/how-to on Pricetaker class usage via an example @MarcusHolly @radhakrishnatg  
- [x] consider testing with IDAES process model (beyond simplified surrogate models that don't use typical IDAES unit models)
- [x] allow for different temporal resolutions and more than one price signal; e.g., cashflow calcs limited to hourly

**February 2025 Release:**
**Finish PR #1358 **
The PriceTaker class must still be merged into the main codebase. This class and associated workflow can be useful for projects related to DISPATCHES-type work. For example, modeling flexible desalination in WaterTAP is of interest, and we could make use of the Pricetaker class.

Here is a to-do list based on @andrewlee94 's suggested changes and @radhakrishnatg 's suggestions for enhancement.

- [x] Add technical reference for Pricetaker class @MarcusHolly #1358 
- [x] Add test file for design_and_operation_model.py @radhakrishnatg 
- [x] Strengthen unit tests across the board (e.g., test `add_startup_shutdown` and other methods); consider simpler dummy cases for unit testing all the methods in isolation @adam-a-a 
- [x] Enable addition of startup, shutdown, ramping, etc. constraints when representative days (as opposed to full-year price signal) are used  
- [x]  Enable addition of cash flow/cost expressions when representative days are used

**Comment by ksbeattie** [comment]
@adam-a-a, @MarcusHolly, @radhakrishnatg will you be opening PRs here in idaes-pse for this?  If so, might one of you be able to join the dev call to discuss and track progress?  If not any of you, then who?

**Comment by adam-a-a** [comment]
@ksbeattie the initial plan is to help get #1358 merged first, and we broke down separate tasks (and assignees) for getting that done. At least one of us should be able to attend every now and then to give updates.

As for new PRs, we would open these here on idaes-PSE, only after merging said PR.

**Comment by MarcusHolly** [comment]
> @ksbeattie the initial plan is to help get #1358 merged first, and we broke down separate tasks (and assignees) for getting that done. At least one of us should be able to attend every now and then to give updates.
> 
> As for new PRs, we would open these here on idaes-PSE, only after merging said PR.

I just joined the call and gave a brief update, more or less saying what you've stated here. I can join the calls in the future.

**Comment by ksbeattie** [comment]
Moving this to the Nov release board.

**Comment by ksbeattie** [comment]
Related to: https://github.com/watertap-org/watertap/pull/1592

We need to identify who will do the remaining check box (first one above).  Perhaps @Xinhe-Chen ?

**Comment by ksbeattie** [comment]
@jakechurchi can you take a pass at this first checkbox here?

**Comment by ksbeattie** [comment]
@radhakrishnatg Do you have approval to put up the NGCC example (for the first checkbox above)?
